POLIO-1794: add dashboard endpoint for subactivities

quang-le opened the pull request:
Add endpoint tailored for ETL use

Related JIRA tickets : POLIO-1794

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [x] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc

comments in code


## Changes

- Add `PaginatedModelViewset`, subclassed from `ModelViewSet`. It enforces the presence of pagination query params when calling `list`

- Add an `/api/polio/dashboard/subactivities` and `/api/polio/dashboard/subactivitiescopes`

## How to test

- Try both endpoints with and without pagination params

## Print screen / video

N/A



## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.

quang-le committed:
POLIO-1794: add PagiatedModelViewSet

- forces GET requests to send pagination params
- Forces use of Paginator for pagination

quang-le committed:
POLIO-1794: add dasboard endpoints for subactivities

- 1 endpoint for SubActivity
- 1 endpoint for SubActivityScope
- tests

tdethier reviewed (changes requested):
What we discussed earlier (400, pagination class name,...)

quang-le committed:
POLIO-1794: refactor PaginatedModelViewset

- rename as EtlModelViewset
- Add default EtlPaginator, with a default `page_size` of 20 and a `max_page_size` of 1000
- Allow overriding paginator if it is a subclass of EtlPaginator, to make sure there's always a `page_size` and a `max_page_size`
- update tests

quang-le committed:
POLIO_1794: add more tests

- Test that `max-page_size` is enforced
- Test EtlModelViewset Paginator subclassing constraint

quang-le committed:
POLIO-1794: remove unused imports

quang-le committed:
POLIO-1794: remove comments

quang-le committed:
POLIO-1794: explicit default max size in test

tdethier reviewed (approved):
LGTM :+1:
